crates/sui-core/src/authority.rs

```diff
@@ -747,10 +748,14 @@ impl AuthorityState {
             epoch_store.reference_gas_price(),
             tx_data,
             input_objects,
-            receiving_objects,
+            &receiving_objects,
             &self.metrics.bytecode_verifier_metrics,
         )?;
 
+        if epoch_store.protocol_config().enable_coin_deny_list() {
```

Comment:
i think it would be better to use this pattern https://github.com/MystenLabs/sui/blob/fc81ab76c60e6f5636cc6754024272fcf5cb3d99/crates/sui-core/src/authority/authority_per_epoch_store.rs#L808-L810, and then in deny_list.rs you can assert that the deny list object exists in `check_coin_deny_list`

Reply:
hmm why is that better though? Right now the system can run properly even when the deny list object hasn't been created yet.